### PR TITLE
 MG-2216 - Rename delete policy function

### LINKED ISSUE #2216: Feature:  Rename `DeletePolicy` gRPC & SpiceDB function  to `DeletePolicyFilter` and
### Is your feature request related to a problem? Please describe.

 `DeletePolicy` gRPC function helps to remove policy with relationship filter, which is different from `DeletePolicies`. 

### Describe the feature you are requesting, as well as the possible use case(s) for it.

 Rename  gRPC, gRPC request & SpiceDB function `DeletePolicy` to `DeletePolicyFilter`

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nyagamunene**
# What type of PR is this?
This is a feature that renames the `DeletePolicy` gRPC function to `DeletePolicyFilter` policy with relationship filter.

## What does this do?
It rename gRPC, gRPC request & SpiceDB function `DeletePolicy` to `DeletePolicyFilter`

## Which issue(s) does this PR fix/relate to?
- Related Issue #2216 
- Resolves #2216 

## Have you included tests for your changes?
Yes.

## Did you document any new/modified feature?
No.

### CODE REVIEW

File: auth/api/logging.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func (lm *loggingMiddleware) AddPolicies(ctx context.Context, prs []auth.PolicyR
 	return lm.svc.AddPolicies(ctx, prs)
 }
 
-func (lm *loggingMiddleware) DeletePolicy(ctx context.Context, pr auth.PolicyReq) (err error) {
+func (lm *loggingMiddleware) DeletePolicyFilter(ctx context.Context, pr auth.PolicyReq) (err error) {
```
Comment (rodneyosodo) [inline auth/api/logging.go:296]: Change the log message

File: auth/api/metrics.go [hunk 2]
```diff
@@ -152,12 +152,12 @@ func (ms *metricsMiddleware) AddPolicies(ctx context.Context, prs []auth.PolicyR
 	return ms.svc.AddPolicies(ctx, prs)
 }
 
-func (ms *metricsMiddleware) DeletePolicy(ctx context.Context, pr auth.PolicyReq) error {
+func (ms *metricsMiddleware) DeletePolicyFilter(ctx context.Context, pr auth.PolicyReq) error {
```
Comment (rodneyosodo) [inline auth/api/metrics.go:155]: Change the name of metric being sent

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: auth/service_test.go [hunk 3]
```diff
@@ -1394,7 +1394,7 @@ func TestDeletePolicies(t *testing.T) {
 	}
 
 	for _, tc := range cases {
-		repocall := prepo.On("DeletePolicies", mock.Anything, mock.Anything).Return(tc.err)
+		repocall := prepo.On("DeletePolicies", context.Background(), mock.Anything).Return(tc.err)
 		err := svc.DeletePolicies(context.Background(), tc.pr)
```
Comment (arvindh123) [inline auth/service_test.go:1398]:
```suggestion
		repocall := prepo.On("DeletePolicyFilter", context.Background(), mock.Anything).Return(tc.err)
```

Comment (arvindh123) [inline auth/service_test.go:1402]: @nyagamunene ,Why we are changing from `mock.Anything` to `context.Background()`

File: auth/policies.go [hunk 4]
```diff
@@ -190,7 +190,7 @@ type PolicyAgent interface {
 	AddPolicies(ctx context.Context, prs []PolicyReq) error
 
 	// DeletePolicy removes a policy.
-	DeletePolicy(ctx context.Context, pr PolicyReq) error
+	DeletePolicyFilter(ctx context.Context, pr PolicyReq) error
```
Comment (arvindh123) [inline auth/policies.go:195]: Please change the comment.

File: auth/policies.go [hunk 5]
```diff
@@ -142,7 +142,7 @@ type Authz interface {
...
```
Comment (arvindh123) [inline auth/policies.go:147]: Please change the comment

File: auth/api/logging.go (hunk 1 above)
Comment (arvindh123) [reply auth/api/logging.go:296]: @nyagamunene Please address this comment

File: auth/api/grpc/endpoint_test.go [hunk 6]
```diff
@@ -470,42 +470,42 @@ func TestDeletePolicy(t *testing.T) {
 	cases := []struct {
 		desc            string
 		token           string
-		deletePolicyReq *magistrala.DeletePolicyReq
-		deletePolicyRes *magistrala.DeletePolicyRes
+		deletePolicyReq *magistrala.DeletePolicyFilterReq
+		deletePolicyRes *magistrala.DeletePolicyFilterRes
 		err             error
```
Comment (arvindh123) [inline auth/api/grpc/endpoint_test.go]:
```suggestion
		deletePolicyFilterReq *magistrala.DeletePolicyFilterReq
		deletePolicyFilterRes *magistrala.DeletePolicyFilterRes
```

File: auth/api/grpc/endpoint_test.go [hunk 7]
```diff
@@ -470,42 +470,42 @@ func TestDeletePolicy(t *testing.T) {
...
 		},
 	}
 	for _, tc := range cases {
-		repoCall := svc.On("DeletePolicy", mock.Anything, mock.Anything).Return(tc.err)
-		dpr, err := client.DeletePolicy(context.Background(), tc.deletePolicyReq)
+		repoCall := svc.On("DeletePolicyFilter", mock.Anything, mock.Anything).Return(tc.err)
+		dpr, err := client.DeletePolicyFilter(context.Background(), tc.deletePolicyReq)
 		assert.Equal(t, tc.deletePolicyRes.GetDeleted(), dpr.GetDeleted(), fmt.Sprintf("%s: expected %v got %v", tc.desc, tc.deletePolicyRes.GetDeleted(), dpr.GetDeleted()))
 		assert.True(t, errors.Contains(err, tc.err), fmt.Sprintf("%s: expected %s got %s\n", tc.desc, tc.err, err))
 		repoCall.Unset()
```
Comment (arvindh123) [inline auth/api/grpc/endpoint_test.go]:
```suggestion
			token: validToken,
			deletePolicyFilterReq: &magistrala.DeletePolicyFilterReq{
				Subject:     id,
				SubjectType: usersType,
				Object:      thingID,
				ObjectType:  thingsType,
				Relation:    readRelation,
				Permission:  readRelation,
			},
			deletePolicyFilterRes: &magistrala.DeletePolicyFilterRes{Deleted: true},
			err:             nil,
		},
		{
			desc:  "delete invalid policy with invalid token",
			token: inValidToken,
			deletePolicyFilterReq: &magistrala.DeletePolicyFilterReq{
				Subject:     id,
				SubjectType: usersType,
				Object:      thingID,
				ObjectType:  thingsType,
				Relation:    readRelation,
				Permission:  readRelation,
			},
			deletePolicyFilterRes: &magistrala.DeletePolicyFilterRes{Deleted: false},
			err:             svcerr.ErrAuthorization,
		},
	}
	for _, tc := range cases {
		repoCall := svc.On("DeletePolicyFilter", mock.Anything, mock.Anything).Return(tc.err)
		dpr, err := client.DeletePolicyFilter(context.Background(), tc.deletePolicyFilterReq)
		assert.Equal(t, tc.deletePolicyFilterRes.GetDeleted(), dpr.GetDeleted(), fmt.Sprintf("%s: expected %v got %v", tc.desc, tc.deletePolicyRes.GetDeleted(), dpr.GetDeleted()))
		assert.True(t, errors.Contains(err, tc.err), fmt.Sprintf("%s: expected %s got %s\n", tc.desc, tc.err, err))
		repoCall.Unset()
```

File: users/service_test.go [hunk 8]
```diff
@@ -970,7 +970,7 @@ func TestUpdateClientRole(t *testing.T) {
 		client               mgclients.Client
 		identifyResponse     *magistrala.IdentityRes
 		authorizeResponse    *magistrala.AuthorizeRes
-		deletePolicyResponse *magistrala.DeletePolicyRes
+		deletePolicyResponse *magistrala.DeletePolicyFilterRes
 		addPolicyResponse    *magistrala.AddPolicyRes
```
Comment (arvindh123) [inline users/service_test.go]:
```suggestion
		deletePolicyFilterResponse *magistrala.DeletePolicyFilterRes
```

File: users/service_test.go [hunk 9]
```diff
@@ -1041,7 +1041,7 @@ func TestUpdateClientRole(t *testing.T) {
 			client:               client2,
 			identifyResponse:     &magistrala.IdentityRes{UserId: client.ID},
 			authorizeResponse:    &magistrala.AuthorizeRes{Authorized: true},
-			deletePolicyResponse: &magistrala.DeletePolicyRes{Deleted: true},
+			deletePolicyResponse: &magistrala.DeletePolicyFilterRes{Deleted: true},
 			updateRoleResponse:   client2,
```
Comment (arvindh123) [inline users/service_test.go]:
```suggestion
			deletePolicyFilterResponse: &magistrala.DeletePolicyFilterRes{Deleted: true},
```

File: users/service_test.go [hunk 10]
```diff
@@ -1050,7 +1050,7 @@ func TestUpdateClientRole(t *testing.T) {
 			desc:                 "update client role to user role with failed to delete policy",
 			identifyResponse:     &magistrala.IdentityRes{UserId: client.ID},
 			authorizeResponse:    &magistrala.AuthorizeRes{Authorized: true},
-			deletePolicyResponse: &magistrala.DeletePolicyRes{Deleted: false},
+			deletePolicyResponse: &magistrala.DeletePolicyFilterRes{Deleted: false},
 			updateRoleResponse:   mgclients.Client{},
```
Comment (arvindh123) [inline users/service_test.go]:
```suggestion
			deletePolicyFilterResponse: &magistrala.DeletePolicyFilterRes{Deleted: false},
```

File: users/service_test.go [hunk 11]
```diff
@@ -1059,7 +1059,7 @@ func TestUpdateClientRole(t *testing.T) {
 			desc:                 "update client role to user role with failed to delete policy with error",
 			identifyResponse:     &magistrala.IdentityRes{UserId: client.ID},
 			authorizeResponse:    &magistrala.AuthorizeRes{Authorized: true},
-			deletePolicyResponse: &magistrala.DeletePolicyRes{Deleted: false},
+			deletePolicyResponse: &magistrala.DeletePolicyFilterRes{Deleted: false},
 			updateRoleResponse:   mgclients.Client{},
```
Comment (arvindh123) [inline users/service_test.go]:
```suggestion
			deletePolicyFilterResponse: &magistrala.DeletePolicyFilterRes{Deleted: false},
```

File: users/service_test.go [hunk 12]
```diff
@@ -1071,7 +1071,7 @@ func TestUpdateClientRole(t *testing.T) {
 			identifyResponse:     &magistrala.IdentityRes{UserId: client.ID},
 			authorizeResponse:    &magistrala.AuthorizeRes{Authorized: true},
 			addPolicyResponse:    &magistrala.AddPolicyRes{Added: true},
-			deletePolicyResponse: &magistrala.DeletePolicyRes{Deleted: true},
+			deletePolicyResponse: &magistrala.DeletePolicyFilterRes{Deleted: true},
 			updateRoleResponse:   mgclients.Client{},
```
Comment (arvindh123) [inline users/service_test.go]:
```suggestion
			deletePolicyFilterResponse: &magistrala.DeletePolicyFilterRes{Deleted: true},
```

File: users/service_test.go [hunk 13]
```diff
@@ -1083,7 +1083,7 @@ func TestUpdateClientRole(t *testing.T) {
 			identifyResponse:     &magistrala.IdentityRes{UserId: client.ID},
 			authorizeResponse:    &magistrala.AuthorizeRes{Authorized: true},
 			addPolicyResponse:    &magistrala.AddPolicyRes{Added: true},
-			deletePolicyResponse: &magistrala.DeletePolicyRes{Deleted: false},
+			deletePolicyResponse: &magistrala.DeletePolicyFilterRes{Deleted: false},
 			updateRoleResponse:   mgclients.Client{},
```
Comment (arvindh123) [inline users/service_test.go]:
```suggestion
			deletePolicyFilterResponse: &magistrala.DeletePolicyFilterRes{Deleted: false},
```

File: auth.proto [hunk 14]
```diff
@@ -116,10 +116,10 @@ message DeletePolicyReq {
 }
 
 message DeletePoliciesReq {
-  repeated DeletePolicyReq deletePoliciesReq = 1;
+  repeated DeletePolicyFilterReq deletePoliciesReq = 1;
 }
```
Comment (arvindh123) [inline auth.proto]: Lets have seperate message for `DeletePoliciesReq`, `DeletePolicyReq` is different from `DeletePolicyFilterReq`
In `DeletePolicies` all fields are required , but in `DeletePolicyFilter` not all the fields are required. 
`DeletePolicyFilter` filters the relation and then delete. 


```suggestion
message DeletePolicyReq {
  string domain = 1;
  string subject_type = 2;
  string subject_relation = 3;
  string subject_kind = 4;
  string subject = 5;
  string relation = 6;
  string permission = 7;
  string object = 8;
  string object_kind = 9;
  string object_type = 10;
}

message DeletePoliciesReq {
  repeated DeletePolicyReq DeletePolicyReq = 1;
}
```

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: auth/service_test.go (hunk 3 above)
Comment (nyagamunene) [reply auth/service_test.go:1402]: When calling `page, err := svc.ListAllObjects(context.Background(), tc.pr)` context.Background is passed to the repo call. This is a mistake that was not noted during the mock test

File: auth.proto [hunk 15]
```diff
@@ -23,7 +23,7 @@ service AuthService {
   rpc Authorize(AuthorizeReq) returns (AuthorizeRes) {}
   rpc AddPolicy(AddPolicyReq) returns (AddPolicyRes) {}
   rpc AddPolicies(AddPoliciesReq) returns (AddPoliciesRes) {}
-  rpc DeletePolicy(DeletePolicyReq) returns (DeletePolicyRes) {}
+  rpc DeletePolicyFilter(DeletePolicyReq) returns (DeletePolicyRes) {}
```
Comment (rodneyosodo) [inline auth.proto]:
```suggestion
  rpc DeletePolicyFilter(DeletePolicyFilterReq) returns (DeletePolicyFilterRes) {}
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: auth/api/metrics.go (hunk 2 above)
Comment (JeffMboya) [reply auth/api/metrics.go:155]: @nyagamunene please address this comment

File: auth.proto (hunk 14 above)
Comment (nyagamunene) [reply auth.proto]: I clarification on this is needed. Should I create a a separate `DeletePolicyFilterRes` and `DeletePolicyFilterReq` to be used?

**Review by rodneyosodo — DISMISSED** [review]
LGTM

**Review by WashingtonKK — DISMISSED** [review]
@nyagamunene I will approve this, however upon testing I found out that `UnshareThing` does not check whether the provided userID is actually a user in the platform. Additionally, unshare thing with a correct thing ID and a random user ID passes successfully, and it can be done multiple times successfully.

Please test this on your end and create an issue for it so that we do not forget it.

**Review by JeffMboya — DISMISSED** [review]

File: auth.proto (hunk 14 above)
Comment (arvindh123) [reply auth.proto]: Yes

File: auth/policies.go [hunk 16]
```diff
@@ -141,8 +141,8 @@ type Authz interface {
 	// only allowed to use as an admin.
 	AddPolicies(ctx context.Context, prs []PolicyReq) error
 
-	// DeletePolicy removes a policy.
-	DeletePolicy(ctx context.Context, pr PolicyReq) error
+	// DeletePolicyFilter removes a policy.
```
Comment (arvindh123) [inline auth/policies.go]: Some thing like below, explain about the function in comments
```go
// DeletePolicyFilter removes policy for given policy filter request.
```

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: auth/api/grpc/endpoint_test.go [hunk 17]
```diff
@@ -468,45 +468,45 @@ func TestDeletePolicy(t *testing.T) {
 	thingID := "thing"
```
Comment (arvindh123) [inline auth/api/grpc/endpoint_test.go:468]: Please change the test function name to `TestDeletePolicyFilter`

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: auth.proto [hunk 18]
```diff
@@ -116,10 +116,10 @@ message DeletePolicyReq {
 }
 
 message DeletePoliciesReq {
-  repeated DeletePolicyReq deletePoliciesReq = 1;
+  repeated DeletePolicyFilterReq deletePoliciesReq = 1;
```
Comment (arvindh123) [inline auth.proto]:
```suggestion

message DeletePolicyFilterReq {
  string domain = 1;
  string subject_type = 2;
  string subject_relation = 3;
  string subject_kind = 4;
  string subject = 5;
  string relation = 6;
  string permission = 7;
  string object = 8;
  string object_kind = 9;
  string object_type = 10;
}
message DeletePolicyFilterRes { bool deleted = 1; }

message DeletePoliciesReq {
  repeated DeletePolicyReq deletePoliciesReq = 1;
}
message DeletePolicyReq {
  string domain = 1;
  string subject_type = 2;
  string subject_relation = 3;
  string subject_kind = 4;
  string subject = 5;
  string relation = 6;
  string permission = 7;
  string object = 8;
  string object_kind = 9;
  string object_type = 10;
}


```

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: auth/service_test.go (hunk 3 above)
Comment (arvindh123) [reply auth/service_test.go:1402]: if we have `mock.Anything` does the test fails

**Review by arvindh123 — DISMISSED** [review]

File: auth/service_test.go (hunk 3 above)
Comment (nyagamunene) [reply auth/service_test.go:1402]: No they don't fail

**Review by arvindh123 — DISMISSED** [review]

**Review by rodneyosodo — DISMISSED** [review]

File: auth/api/grpc/endpoint_test.go [hunk 19]
```diff
@@ -468,45 +468,45 @@ func TestDeletePolicy(t *testing.T) {
...
-			err:             svcerr.ErrAuthorization,
+			deletePolicyFilterRes: &magistrala.DeletePolicyFilterRes{Deleted: false},
+			err:                   svcerr.ErrAuthorization,
 		},
 	}
 	for _, tc := range cases {
-		repoCall := svc.On("DeletePolicy", mock.Anything, mock.Anything).Return(tc.err)
-		dpr, err := client.DeletePolicy(context.Background(), tc.deletePolicyReq)
-		assert.Equal(t, tc.deletePolicyRes.GetDeleted(), dpr.GetDeleted(), fmt.Sprintf("%s: expected %v got %v", tc.desc, tc.deletePolicyRes.GetDeleted(), dpr.GetDeleted()))
+		repoCall := svc.On("DeletePolicyFilter", mock.Anything, mock.Anything).Return(tc.err)
```
Comment (JeffMboya) [inline auth/api/grpc/endpoint_test.go]:
```suggestion
		svcCall := svc.On("DeletePolicyFilter", mock.Anything, mock.Anything).Return(tc.err)
```

File: auth/service_test.go [hunk 20]
```diff
@@ -1448,7 +1448,7 @@ func TestListObjects(t *testing.T) {
 		},
 	}
 	for _, tc := range cases {
-		repocall2 := prepo.On("RetrieveObjects", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(expectedPolicies, mock.Anything, tc.err)
+		repocall2 := prepo.On("RetrieveObjects", context.Background(), mock.Anything, mock.Anything, mock.Anything).Return(expectedPolicies, mock.Anything, tc.err)
```
Comment (JeffMboya) [inline auth/service_test.go:1451]:
```suggestion
		repocall := prepo.On("RetrieveObjects", context.Background(), mock.Anything, mock.Anything, mock.Anything).Return(expectedPolicies, mock.Anything, tc.err)
```

File: auth/service_test.go [hunk 21]
```diff
@@ -1516,7 +1516,7 @@ func TestCountObjects(t *testing.T) {
 
 	pageLen := uint64(15)
 
-	repocall2 := prepo.On("RetrieveAllObjectsCount", mock.Anything, mock.Anything, mock.Anything).Return(pageLen, nil)
+	repocall2 := prepo.On("RetrieveAllObjectsCount", context.Background(), mock.Anything, mock.Anything).Return(pageLen, nil)
```
Comment (JeffMboya) [inline auth/service_test.go:1519]:
```suggestion
	repocall := prepo.On("RetrieveAllObjectsCount", context.Background(), mock.Anything, mock.Anything).Return(pageLen, nil)
```

File: auth/service_test.go [hunk 22]
```diff
@@ -1566,7 +1566,7 @@ func TestListSubjects(t *testing.T) {
 		},
 	}
 	for _, tc := range cases {
-		repocall2 := prepo.On("RetrieveSubjects", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(expectedPolicies, mock.Anything, tc.err)
+		repocall2 := prepo.On("RetrieveSubjects", context.Background(), mock.Anything, mock.Anything, mock.Anything).Return(expectedPolicies, mock.Anything, tc.err)
```
Comment (JeffMboya) [inline auth/service_test.go]: same

File: auth/service_test.go [hunk 23]
```diff
@@ -1619,7 +1619,7 @@ func TestListAllSubjects(t *testing.T) {
 		},
 	}
 	for _, tc := range cases {
-		repocall2 := prepo.On("RetrieveAllSubjects", mock.Anything, mock.Anything).Return(expectedPolicies, tc.err)
+		repocall2 := prepo.On("RetrieveAllSubjects", context.Background(), mock.Anything).Return(expectedPolicies, tc.err)
```
Comment (JeffMboya) [inline auth/service_test.go]: same

File: auth/service_test.go [hunk 24]
```diff
@@ -1653,7 +1653,7 @@ func TestListPermissions(t *testing.T) {
 	}
 	filterPermisions := []string{auth.ViewPermission, auth.AdminPermission}
 
-	repoCall1 := prepo.On("RetrievePermissions", mock.Anything, pr, filterPermisions).Return(auth.Permissions{}, nil)
+	repoCall1 := prepo.On("RetrievePermissions", context.Background(), pr, filterPermisions).Return(auth.Permissions{}, nil)
```
Comment (JeffMboya) [inline auth/service_test.go]:
```suggestion
	repoCall := prepo.On("RetrievePermissions", context.Background(), pr, filterPermisions).Return(auth.Permissions{}, nil)
```

File: internal/groups/service_test.go [hunk 25]
```diff
@@ -2468,16 +2468,16 @@ func TestDeleteGroup(t *testing.T) {
...
 				Deleted: true,
 			},
-			deleteDomainsPoliciesRes: &magistrala.DeletePolicyRes{
+			deleteDomainsPoliciesRes: &magistrala.DeletePolicyFilterRes{
 				Deleted: true,
 			},
-			deleteUsersPoliciesRes: &magistrala.DeletePolicyRes{
+			deleteUsersPoliciesRes: &magistrala.DeletePolicyFilterRes{
 				Deleted: true,
 			},
```
Comment (JeffMboya) [inline internal/groups/service_test.go:2511]:
```suggestion
	authzResp: &magistrala.AuthorizeRes{Authorized: true},
    deleteChildPoliciesRes: &magistrala.DeletePolicyFilterRes{Deleted: true},
    deleteThingsPoliciesRes: &magistrala.DeletePolicyFilterRes{Deleted: true},
    deleteDomainsPoliciesRes: &magistrala.DeletePolicyFilterRes{Deleted: true},
    deleteUsersPoliciesRes: &magistrala.DeletePolicyFilterRes{Deleted: true},
```

Comment (JeffMboya) [reply internal/groups/service_test.go:2511]: same as all other cases

File: internal/groups/service_test.go [hunk 26]
```diff
@@ -2626,23 +2626,23 @@ func TestDeleteGroup(t *testing.T) {
 				Object:      tc.groupID,
 				ObjectType:  auth.GroupType,
 			}).Return(tc.authzResp, tc.authzErr)
-			repocall2 := authsvc.On("DeletePolicy", context.Background(), &magistrala.DeletePolicyReq{
+			repocall2 := authsvc.On("DeletePolicyFilter", context.Background(), &magistrala.DeletePolicyFilterReq{
```
Comment (JeffMboya) [inline internal/groups/service_test.go]:
```suggestion
			authCallll2 := authsvc.On("DeletePolicyFilter", context.Background(), &magistrala.DeletePolicyFilterReq{
```

Comment (JeffMboya) [reply internal/groups/service_test.go]: same as other cases

File: pkg/sdk/go/channels_test.go [hunk 27]
```diff
@@ -769,7 +769,7 @@ func TestDeleteChannel(t *testing.T) {
 	repoCall1.Unset()
 	repoCall2.Unset()
 
-	repoCall = auth.On("DeletePolicy", mock.Anything, mock.Anything, mock.Anything).Return(&magistrala.DeletePolicyRes{Deleted: true}, nil)
+	repoCall = auth.On("DeletePolicyFilter", mock.Anything, mock.Anything, mock.Anything).Return(&magistrala.DeletePolicyFilterRes{Deleted: true}, nil)
```
Comment (JeffMboya) [inline pkg/sdk/go/channels_test.go]: same

File: pkg/sdk/go/groups_test.go [hunk 28]
```diff
@@ -909,7 +909,7 @@ func TestDeleteGroup(t *testing.T) {
 	repoCall1.Unset()
 	repoCall2.Unset()
 
-	repoCall = auth.On("DeletePolicy", mock.Anything, mock.Anything, mock.Anything).Return(&magistrala.DeletePolicyRes{Deleted: true}, nil)
+	repoCall = auth.On("DeletePolicyFilter", mock.Anything, mock.Anything, mock.Anything).Return(&magistrala.DeletePolicyFilterRes{Deleted: true}, nil)
```
Comment (JeffMboya) [inline pkg/sdk/go/groups_test.go]: same

**Review by JeffMboya — CHANGES_REQUESTED** [review]

File: internal/groups/service_test.go (hunk 25 above)
Comment (nyagamunene) [reply internal/groups/service_test.go:2511]: I think the format we currently have is much better.

**Review by JeffMboya — DISMISSED** [review]

**Review by arvindh123 — DISMISSED** [review]

**Review by JeffMboya — DISMISSED** [review]

**Review by arvindh123 — DISMISSED** [review]

**Comment by JeffMboya** [comment]
> @nyagamunene I will approve this, however upon testing I found out that `UnshareThing` does not check whether the provided userID is actually a user in the platform. Additionally, unshare thing with a correct thing ID and a random user ID passes successfully, and it can be done multiple times successfully.
> 
> Please test this on your end and create an issue for it so that we do not forget it.

@nyagamunene @WashingtonKK for unshare operation, the policy is deleted for all users (whether the user is part of magistrala or non-existent/random) by design

**Review by arvindh123 — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**